openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +753,66 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def update_houdini_vars_context():
+    """Update Houdini vars to match current context.
+
+    This will only do something if the setting is enabled in project settings.
+    """
+
+    project_settings = get_current_project_settings()
+    houdini_vars_settings = \
+        project_settings["houdini"]["general"]["update_houdini_var_context"]
+
+    if not houdini_vars_settings["enabled"]:
+        return
+
+    houdini_vars = houdini_vars_settings["houdini_vars"]
+
+    # No vars specified - nothing to do
+    if not houdini_vars:
+        return
+
+    # Get Template data
+    template_data = get_current_context_template_data()
+
+    # Set Houdini Vars
+    for item in houdini_vars:
```

Comment:
Is `houdini_vars` a dictionary? If so we could do:
`for var_name, var_value in houdini_vars.items():`